packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -267,12 +267,43 @@ class SystemChrome {
   ///
   /// If a particular overlay is unsupported on the platform, enabling or
   /// disabling that overlay will be ignored.
+  /// 
+  /// The settings here can be overidden by the platform when System UI becomes
+  /// nessecary for functionality.
+  /// 
+  /// On Android, when the keyboard becomes visible, it will enable navigation bar
+  /// and status bar system UI overlays. When the keyboard is closed, Android will
+  /// not restore the previous UI visibility settings, and the UI visibility cannot
+  /// be changed until 1 second after the keyboard is closed to prevent malware 
+  /// locking users from navigation buttons.
+  /// 
+  /// To regain "fullscreen" after text entry, the UI overlays should be set again
+  /// after a delay of 1 second. This can be achieved through [restoreSystemUIOverlays]
+  /// or calling this again. Otherwise, the original UI overlay settings will be
+  /// automatically restored only when the application loses and regains focus.
   static Future<Null> setEnabledSystemUIOverlays(List<SystemUiOverlay> overlays) async {
     await SystemChannels.platform.invokeMethod(
       'SystemChrome.setEnabledSystemUIOverlays',
       _stringify(overlays),
     );
- }
+  }
+
+  /// Restores the system overlays to the settings provided in [setEnabledSystemUIOverlays].
+  /// 
+  /// May be used when the platform force enables/disables UI elements. This will
+  /// restore the previous settings as defined by [setEnabledSystemUIOverlays].
+  /// 
+  /// For example, when the Android keyboard disables hidden status and navigation bars,
+  /// this can be called to re-disable the bars when the keyboard is closed.
+  /// 
+  /// On Android, the system UI cannot be changed until 1 second after the previous change.
+  /// This is to prevent malware from permanently hiding navigation buttons.
```

Comment:
Maybe lead with: `Note: On Android ...`

Similar comment to above about commenting on the rationale for this behaviour.

Reply:
Ah, I just gave the opposite comment!

We try to avoid "Note" and "Note that" and so on in the docs, since it doesn't add anything.